aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -395,6 +396,15 @@ static void clamp_min_kernel(TensorIterator& iter, Scalar min_scalar) {
   });
 }
 
+static void kaiser_window_kernel(TensorIterator& iter, int64_t window_length, double beta){
+  AT_DISPATCH_FLOATING_TYPES_AND(kBFloat16, iter.dtype(), "kaiser_window_cpu", [&](){
+    auto alpha = (window_length - 1) / 2.0;
+    cpu_kernel(iter, [=](scalar_t a){
+        return static_cast<scalar_t>(calc_i0(beta * std::sqrt(1 - std::pow((a - alpha) / alpha, 2.0))) / calc_i0(beta));
```

Comment:
Can you elaborate on this formulation of the window?

The NumPy documentation of the window appears to be incorrect. The NumPy implementation is:

https://github.com/numpy/numpy/blob/54811204a63c2606e89c4f053c4b9a7215207079/numpy/lib/function_base.py#L3328

and the documentation is [here](https://numpy.org/doc/stable/reference/generated/numpy.kaiser.html?highlight=kaiser#numpy.kaiser).

This follows the NumPy implementation and the [Mathworks documentation](https://www.mathworks.com/help/signal/ref/kaiser.html), which is equivalent to [Wikipedia's description](https://en.wikipedia.org/wiki/Kaiser_window) when "symmetrically sampling" the window. 

In this case `a` is a positive integer and `alpha` is always a positive float >= 1, is it actually preferable to divide `a - alpha` by `alpha` or just to divide `a` by `alpha` and subtract one afterwards?

Reply:
Yes, you are right, the numpy and scipy documentation for this function is wrong. I copied their function documentation, but I'll change it to the Mathworks documentation.

I can change it to divide first, then subtract one. I'm not sure if it will really make a difference.

Reply:
I am sorry, I was mistaken - their documentation is actually correct. You actually found the key point in a later comment, with the range that their n is defined in. Numpy shifted the the range leftwards by the N/2 term in the mathworks docs, so that the range for n is -N/2 to N/2, while in mathworks it is 0 to N. In the numerator in mathworks, they do n-N/2, but since numpy has already shifted their range, they do not need to subtract anything. The extra 4 in the numerator in numpy is from the the mathworks denominator - they square (N/2) in the denominator, while numpy only squares N in the denominator.

Let me know which formula is preferred for the docs.

Reply:
Oh that's really interesting. Thank you for explaining this discrepancy to me. As mentioned below I really struggled with the way NumPy was presenting the range of the value. I think taking a couple points from your explanation here would be helpful. I would stick with what makes the most sense to you.